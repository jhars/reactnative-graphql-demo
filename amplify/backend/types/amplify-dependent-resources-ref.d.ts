export type AmplifyDependentResourcesAttributes = {
  "auth": {
    "laxappdemo451d064d": {
      "AppClientID": "string",
      "AppClientIDWeb": "string",
      "IdentityPoolId": "string",
      "IdentityPoolName": "string",
      "UserPoolArn": "string",
      "UserPoolId": "string",
      "UserPoolName": "string"
    }
  }
}