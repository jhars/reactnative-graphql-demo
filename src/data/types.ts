export type ById<T> = { [index: string]: T };

export interface DropPlayerRequestObject {
  leagueId: number;
  playerId: number;
  rosterSpot: RosterSpot;
  rosterId: string;
  playerInfo: Player;
}

export interface SelectDropPlayerObject {
  leagueId: number;
  playerId: number;
  rosterSpot: RosterSpot;
  rosterId: string;
  playerInfo: Player;
}

export enum SortColumnName {
  Name = "Name",
  Points = "Points",
  Position = "Position",
}

export enum Position {
  G = "G",
  D = "D",
  LSM = "LSM",
  SSDM = "SSDM",
  FO = "FO",
  M = "M",
  A = "A",
}

export enum RosterSpot {
  GOALIE = "GOALIE",
  DEFENSE1 = "DEFENSE1",
  DEFENSE2 = "DEFENSE2",
  LSM = "LSM",
  SSDM = "SSDM",
  FO = "FO",
  MIDFIELD1 = "MIDFIELD1",
  MIDFIELD2 = "MIDFIELD2",
  ATTACK1 = "ATTACK1",
  ATTACK2 = "ATTACK2",
  BENCH1 = "BENCH1",
  BENCH2 = "BENCH2",
  BENCH3 = "BENCH3",
  BENCH4 = "BENCH4",
}

interface StatLine {
  points: number;
  goals: number;
  assists: number;
  turnovers: number;
  groundBalls: number;
  faceoffsWon: number;
  saves: number;
}

interface Statistics {
  statLineLastSeason: StatLine;
}

export interface Player {
  id: string;
  firstName: string;
  lastName: string;
  statistics: Statistics;
  position: Position;
}

export interface PlayerData {
  players: Player[];
}

export interface Players {
  data: PlayerData;
}

export interface League {
  id: string;
  title: string;
  teams: Team[];
}

export interface LeaguesData {
  leagues: League[];
}

// interface Leagues {
//   data: LeaguesData;
// }

export interface Team {
  id: string;
  name: string;
  league: League;
}

export interface TeamsData {
  teams: Team[];
}

// interface Teams {
//   data: TeamsData;
// }

export interface Roster {
  id: string;
  teamId: number;
  teamInfo: Team;
  goalie: Player;
  defense1: Player;
  defense2: Player;
  lsm: Player;
  ssdm: Player;
  fo: Player;
  midfield1: Player;
  midfield2: Player;
  attack1: Player;
  attack2: Player;
}

export interface RosterData {
  roster: Roster;
}

interface AddPlayerToTeamRosterData {
  addPlayerToTeamRoster: Roster;
}

export interface AddPlayerToTeamRosterMutationResponse {
  data: AddPlayerToTeamRosterData;
}

//============================
//============================
export interface CurrentUser {
  id: string;
  email: string;
  preferred_username: string;
}
