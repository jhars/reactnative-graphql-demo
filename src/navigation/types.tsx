import { RouteProp, NavigatorScreenParams, ParamListBase, CompositeScreenProps, CompositeNavigationProp } from '@react-navigation/native';
import { StackNavigationProp } from '@react-navigation/stack';
import {DrawerNavigationProp} from '@react-navigation/drawer';
import { Team, Position, RosterSpot, League } from '../data/types';
import type { NativeStackScreenProps, NativeStackNavigationProp } from '@react-navigation/native-stack';

//JH-NOTE: review all of this... necessary? 
export type BaseNavigationProps = NativeStackNavigationProp<ParamListBase>;

export type AvailablePlayersScreenRouteProp = RouteProp<MyTeamsStackParamList, 'AvailablePlayersScreen'>;

export type MyTeamsScreenNavigationProps = NavigatorScreenParams<MyTeamsStackParamList>;

export type LeagueTeamsScreenRouteProps = RouteProp<LeaguesStackParamList, 'LeagueTeamsScreen'>;
export type LeaguesScreenNavigationProps = NavigatorScreenParams<LeaguesStackParamList>;

export type CreateNewTeamConfirmationScreenRouteProp = RouteProp<CreateTeamStackParamList, 'CreateNewTeamConfirmation'>;
export type CreateNewTeamNavigationProps = NavigatorScreenParams<CreateTeamStackParamList>;

export type RosterScreenRouteProps = RouteProp<LeaguesStackParamList|MyTeamsStackParamList,'RosterScreen'>;
// export type RosterScreenRouteProps = RouteProp<ParamListBase>;
// export type RosterScreenRouteProps = NavigatorScreenParams<RosterScreenParamList>;
// export type RosterScreenRouteProps = RouteProp<ParamListBase>;
// export type RosterScreenRouteProps = NativeStackNavigationProp<RosterScreenParamList>;

// export type RosterScreenCompositeProps = CompositeNavigationProp<
//   NativeStackNavigationProp<ParamListBase, T>,
//   RouteProp<RosterScreenParamList>
// >;

// export type RootStackScreenProps<T extends keyof RootDrawerParamList> = StackScreenProps<RootDrawerParamList, T>;


// export type RosterScreenCompositeProps<T extends keyof RosterScreenParamList> =
//   CompositeScreenProps<
//     RouteProp<ParamListBase, T>,
//     RootStackScreenProps<keyof RootStackParamList>
//   >;



export type RootDrawerParamList = {
  Home: undefined;
  MyTeams: MyTeamsScreenNavigationProps;
  Leagues: LeaguesScreenNavigationProps;
  Players: undefined;
  Account: undefined;
  CreateTeam: CreateNewTeamNavigationProps;
};

export type CreateTeamStackParamList = { 
  SelectLeagueFromList: undefined,
  CreateNewTeamConfirmation: {
    leagueID: number;
    leagueTitle: string;
  }
}

export type RosterScreenParamList = {
  team: Team | undefined;
  myTeam: boolean | undefined;
}

export type MyTeamsStackParamList = {
  UserTeams: undefined,
  RosterScreen: RosterScreenParamList,
  AvailablePlayersScreen: { 
    availableForLeagueId: number | undefined;
    team: Team | undefined;
    position: Position | undefined;
    rosterId: string | undefined;
    rosterSpot: RosterSpot | undefined;
  }
};

export type LeaguesStackParamList = {
  LeaguesScreen: undefined,
  LeagueTeamsScreen: {
    league: League;
  },
  RosterScreen: RosterScreenParamList
};



declare global {
  namespace ReactNavigation {
    interface RootParamList extends RootDrawerParamList {}
  }
}
