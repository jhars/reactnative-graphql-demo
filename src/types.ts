export type ById<T> = { [index: string]: T };

interface StatLine {
  points: number;
  goals: number;
  assists: number;
  turnovers: number;
  groundBalls: number;
  faceoffsWon: number;
  saves: number;
}

interface Statistics {
  statLineLastSeason: StatLine;
}

interface Player {
  id: string;
  firstName: string;
  lastName: string;
  jersey: number;
  statistics: Statistics;
}

interface PlayerData {
  players: Player[];
}

export interface Players {
  data: PlayerData;
}

interface League {
  id: serial;
  title: string;
  teams: Team[];
}

interface LeagueData {
  leagues: League[];
}

interface Leagues {
  data: LeagueData;
}

export interface Team {
  id: string;
  name: string;
}

interface TeamData {
  teams: Team[]
}

export interface Teams {
  data: TeamData
}

export interface Roster {
  teamId: number;
  goalie: Player;
  defense1: Player;
  defense2: Player;
  lsm: Player;
  fo: Player;
  midfield1: Player;
  midfield2: Player;
  attack1: Player;
  attack2: Player;
}

interface RosterData {
  roster: Roster;
}

export interface Roster {
  data: RosterData;
}

//============================
//============================
export interface CurrentUser {
  id: serial;
  email: string;
}