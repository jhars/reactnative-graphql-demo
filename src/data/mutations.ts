import { gql } from '@apollo/client';

export const ADD_TEAM_TO_LEAGUE = gql`
	mutation AddTeam($name: String!, $leagueId: Int!, $ownerId: ID!) {
	  addTeam(name: $name, leagueId: $leagueId, ownerId: $ownerId) {
	    name
	    leagueId
	    ownerId
	    league {
	      title
	      id
	    }
	  }
	}
`;
export const ADD_PLAYER_TO_TEAM = gql`
mutation AddPlayerToTeamRoster($leagueId: Int!, $teamId: Int!, $playerId: Int!, $position: Position!, $rosterSpot: RosterSpot!,$rosterId: ID!) {
  addPlayerToTeamRoster(leagueId: $leagueId, teamId: $teamId, playerId: $playerId, position: $position, rosterSpot: $rosterSpot, rosterId: $rosterId) {

      goalie {
        lastName
        position
        statistics {
          statLineLastSeason {
            points
          }
        }
      }

      defense1 {
        lastName
        position
        statistics {
          statLineLastSeason {
            points
          }
        }
      }
    
      defense2 {
        lastName
        position
        statistics {
          statLineLastSeason {
            points
          }
        }
      }

      lsm {
        lastName
        position
        statistics {
          statLineLastSeason {
            points
          }
        }
      }

      fo {
        lastName
        position
        statistics {
          statLineLastSeason {
            points
          }
        }
      }

      ssdm {
        lastName
        position
        statistics {
          statLineLastSeason {
            points
          }
        }
      }

      midfield1 {
        lastName
        position
        statistics {
          statLineLastSeason {
            points
          }
        }
      }

      midfield2 {
        lastName
        position
        statistics {
          statLineLastSeason {
            points
          }
        }
      }

      attack1 {
        lastName
        position
        statistics {
          statLineLastSeason {
            points
          }
        }
      }

      attack2 {
        lastName
        position
        statistics {
          statLineLastSeason {
            points
          }
        }
      } 
    }

}
`;