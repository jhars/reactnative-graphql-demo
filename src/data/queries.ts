import { gql } from '@apollo/client';

export const GET_SORTED_PLAYERS = gql`
  query Players($orderBy: PlayerOrder!, $position: Position, $availableForLeagueId: Int) {
    players(orderBy: $orderBy, position: $position, availableForLeagueId: $availableForLeagueId) {
      id
      lastName
      position
      statistics {
        statLineLastSeason {
          points
        }
      }
    }
  }
`;

export const GET_ALL_LEAGUES = gql`
	query Leagues {
	  leagues {
	    id
	    title
	    teams {
	      id
	      name
	    }
	  }
	}
`;

export const GET_TEAMS = gql`
  query Teams($ownerId: ID, $teamsId: Int, $leagueId: Int) {
    teams(ownerId: $ownerId, teamsId: $teamsId, leagueId: $leagueId) {
      id
      name
      league {
        title
        id
      }
    }
  }
`;



export const GET_TEAM_ROSTER = gql`
query Roster($teamId: Int) {
  roster(teamId: $teamId) {

    id
    
    teamInfo {
      name
      id
      league {
        title
        id
      }
    }

    goalie {
      id
      lastName
      position
      statistics {
        statLineLastSeason {
          points
          saves
        }
      }
    }

    defense1 {
    id
    lastName
    position
    statistics {
      statLineLastSeason {
        points
        turnovers
      }
     } 
    }
   
    defense2 {
      id
      lastName
      position
      statistics {
        statLineLastSeason {
          points
          turnovers
      }
     } 
    }

    lsm {
      id
      lastName
      position
      statistics {
        statLineLastSeason {
          points
          groundBalls
        }
      }
    }

    ssdm {
      id
      lastName
      position
      statistics {
        statLineLastSeason {
          points
          groundBalls
        }
      }
    }

    fo {
      id
      lastName
      position
      statistics {
        statLineLastSeason {
          points
          faceoffsWon
        }
      }
    }

    midfield1 {
      id
      lastName
      position
      statistics {
        statLineLastSeason {
          points
          goals
          assists
        }
      }
    }

    midfield2 {
      id
      lastName
      position
      statistics {
        statLineLastSeason {
          points
          goals
          assists
        }
      }
    }

    attack1 {
      id
      lastName
      position
      statistics {
        statLineLastSeason {
          points
          goals
          assists
        }
      }
    }

    attack2 {
      id
      lastName
      position
      statistics {
        statLineLastSeason {
          points
          goals
          assists
        }
      }
    } 
  }
}
`;