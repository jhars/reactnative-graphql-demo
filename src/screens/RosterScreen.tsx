import React, {useEffect} from 'react';
import { StyleSheet, ActivityIndicator, Text } from 'react-native';
import RosterList from '../components/RosterList';
import { useNavigation } from '@react-navigation/native';

//==== GraphQL ========
import { useQuery } from '@apollo/client';
import { GET_TEAM_ROSTER } from '../data/queries';
import { Teams } from '../types';
//=====================

export default function RosterScreen({route}) {
  const navigation = useNavigation();
  useEffect(()=>{
    navigation.setOptions({
          title: route.params.team.name,
        });
  }, []);

  //JH-NOTE: Query with parameters here...
  const { loading, error, data } = useQuery<Roster>(GET_TEAM_ROSTER, {
    variables: { teamId: Number(route.params.team.id) }
  });

  if (loading) return <ActivityIndicator testID="loading" size="large" color="#0000ff" />;
  if (error) return <Text>Error: {error.message}</Text>;

  const roster = data?.roster
  console.log("roster: ")
  console.log(roster)
  console.log("========")

  return(
    <RosterList roster={roster}/>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#fff',
    alignItems: 'center',
    justifyContent: 'center',
  },
});
