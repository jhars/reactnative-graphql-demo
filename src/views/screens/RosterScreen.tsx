import React, {useEffect, useState, useCallback} from 'react';
import { useNavigation, useFocusEffect, useRoute } from '@react-navigation/native';
import { Alert, View, Text, StyleSheet, ScrollView, Modal, ActivityIndicator} from 'react-native';
import RosterTableHeader from '../components/RosterTableHeader';
import RosterRow from '../components/RosterRow';
import DropPlayerConfirmationModal from '../components/DropPlayerConfirmationModal'
//==== GraphQL ========
import { useQuery } from '@apollo/client';
import { GET_TEAM_ROSTER } from '../../data/queries';
import { RosterData, DropPlayerRequestObject, RosterSpot, Position } from '../../data/types';
//=====================
import { RosterScreenRouteProps } from '../../navigation/types';

export default function RosterScreen() {
  const navigation = useNavigation();
  const route = useRoute<RosterScreenRouteProps>();

  const [ dropPlayer, setDropPlayer ] = useState<DropPlayerRequestObject|null>(null);
  const [ modalVisible, setModalVisible ] = useState(false);

  const { team, myTeam } = route.params

  useEffect(()=>{
    navigation.setOptions({
          title: team?.name,
        });
  }, []);

  useFocusEffect(
    useCallback(() => {
      refetch()
    }, [])
  )

  const selectPlayerToDrop = (player: DropPlayerRequestObject) => {
    setDropPlayer(player)
    setModalVisible(true)
  }
  
  //JH-NOTE: may want to define type here...
  const dropPlayerFromRosterRowCallback = useCallback( (player: DropPlayerRequestObject) => {
    selectPlayerToDrop(player)
  }, []);

  const modalCallback = useCallback( () => {
    refetch()
    setModalVisible(false)
  }, []);

  const cancelModalCallback = useCallback( () => {
    setModalVisible(false)
  }, []);

  const failedToDropPlayerCallback = useCallback( () => {
    showFailureAlert()
  }, []);

  const showFailureAlert = () => {
    setModalVisible(false)
    Alert.alert('Unable to Drop Player','Player may have started games this week or there may be network issues', [
      {
        text: 'OK'
      }
    ])
  } 

  const { loading, error, data, refetch } = useQuery<RosterData>(GET_TEAM_ROSTER, {
    variables: { teamId: Number(team?.id) }
  });

  const roster = data?.roster

  if (loading) return <ActivityIndicator testID="loading" size="large" color="#0000ff" />;
  if (error) return <Text>Error: {error.message}</Text>;

    return (
      <View style={styles.container}>
        <RosterTableHeader myTeam={myTeam}/>
          <Modal
            animationType="slide"
            transparent={true}
            visible={modalVisible}
            onRequestClose={() => setModalVisible(!modalVisible)}>

            {(dropPlayer && roster) &&
              <DropPlayerConfirmationModal 
                rosterSpot={dropPlayer.rosterSpot}
                playerId={dropPlayer.playerInfo.id}
                lastName={dropPlayer.playerInfo.lastName}
                position={dropPlayer.playerInfo.position}
                roster={roster}
                modalCallback={modalCallback}
                cancelCallback={cancelModalCallback}
                failedToDropPlayerCallback={failedToDropPlayerCallback}/>
            }

          </Modal>
        
          {roster &&
          <ScrollView>
            <RosterRow dropPlayerCallback={dropPlayerFromRosterRowCallback} rosterId={roster.id} rosterSpot={RosterSpot.GOALIE} position={Position.G} playerInfo={roster.goalie} myTeam={myTeam} teamInfo={roster?.teamInfo}/>
            <RosterRow dropPlayerCallback={dropPlayerFromRosterRowCallback} rosterId={roster.id} rosterSpot={RosterSpot.DEFENSE1} position={Position.D} playerInfo={roster.defense1} myTeam={myTeam} teamInfo={roster?.teamInfo}/>
            <RosterRow dropPlayerCallback={dropPlayerFromRosterRowCallback} rosterId={roster.id} rosterSpot={RosterSpot.DEFENSE2} position={Position.D} playerInfo={roster.defense2} myTeam={myTeam} teamInfo={roster?.teamInfo}/>
            <RosterRow dropPlayerCallback={dropPlayerFromRosterRowCallback} rosterId={roster.id} rosterSpot={RosterSpot.LSM} position={Position.LSM} playerInfo={roster.lsm} myTeam={myTeam} teamInfo={roster?.teamInfo}/>
            <RosterRow dropPlayerCallback={dropPlayerFromRosterRowCallback} rosterId={roster.id} rosterSpot={RosterSpot.SSDM} position={Position.SSDM} playerInfo={roster.ssdm} myTeam={myTeam} teamInfo={roster?.teamInfo}/>
            <RosterRow dropPlayerCallback={dropPlayerFromRosterRowCallback} rosterId={roster.id} rosterSpot={RosterSpot.FO} position={Position.FO} playerInfo={roster.fo} myTeam={myTeam} teamInfo={roster?.teamInfo}/>
            <RosterRow dropPlayerCallback={dropPlayerFromRosterRowCallback} rosterId={roster.id} rosterSpot={RosterSpot.MIDFIELD1} position={Position.M} playerInfo={roster.midfield1} myTeam={myTeam} teamInfo={roster?.teamInfo}/>
            <RosterRow dropPlayerCallback={dropPlayerFromRosterRowCallback} rosterId={roster.id} rosterSpot={RosterSpot.MIDFIELD2} position={Position.M} playerInfo={roster.midfield2} myTeam={myTeam} teamInfo={roster?.teamInfo}/>
            <RosterRow dropPlayerCallback={dropPlayerFromRosterRowCallback} rosterId={roster.id} rosterSpot={RosterSpot.ATTACK1} position={Position.A} playerInfo={roster.attack1} myTeam={myTeam} teamInfo={roster?.teamInfo}/>
            <RosterRow dropPlayerCallback={dropPlayerFromRosterRowCallback} rosterId={roster.id} rosterSpot={RosterSpot.ATTACK2} position={Position.A} playerInfo={roster.attack2} myTeam={myTeam} teamInfo={roster?.teamInfo}/>
          </ScrollView>   
          }
          
        
      </View>
    );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#fff',
  }
});
