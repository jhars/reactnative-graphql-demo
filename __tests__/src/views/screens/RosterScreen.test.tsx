/**
 * @format
 */

import 'react-native';
import React from 'react';
import RosterScreen from '../../../../src/views/screens/RosterScreen';
import {it, describe} from '@jest/globals';
import { render, screen, waitFor } from '@testing-library/react-native';
import {Roster} from '../../../../src/data/types';
import { MOCK_TEAM, MOCK_ROSTER } from '../../../mocks/MOCKS_SHARED'
import {MockedProvider} from '@apollo/react-testing';
import { GET_TEAM_ROSTER } from '../../../../src/data/queries';
import { NavigationContainer } from '@react-navigation/native';

jest.mock("@react-navigation/native", () => {
  const actualNav = jest.requireActual("@react-navigation/native");
  return {
    ...actualNav,
    useNavigation: () => ({
      navigate: jest.fn(),
      setOptions: jest.fn(() => {title: MOCK_TEAM.name})
    }),
    useRoute: () => ({
    	params: {
    		myTeam: false,
    		team: MOCK_TEAM
    	}
    })
  };
});

describe('Roster Screen Tests', () => {
	
	it('ASSERT SOMETHING HERE', async () => {
		const mockRoster: Roster = MOCK_ROSTER
		const mockTeamId = MOCK_TEAM.id

		const mocks = [
		  {
		    request: {
		      query: GET_TEAM_ROSTER,
		      variables: { teamId: Number(mockTeamId) }
		    },
		    result: {
		    	data: {
		    		roster: MOCK_ROSTER
		    	}
		    }
		  }
		];

	  const Roster = render(
	  	<MockedProvider mocks={mocks} addTypename={false}>
	  		<NavigationContainer>
  				<RosterScreen />
  			</NavigationContainer>
	  	</MockedProvider>
	  )

	  const playerName = await screen.findByText('McStickens')
	  const testedTree = screen.toJSON()

    await waitFor(() => {
    	expect(playerName).toBeVisible()
      expect(testedTree).toMatchSnapshot()
    });
	});

});


