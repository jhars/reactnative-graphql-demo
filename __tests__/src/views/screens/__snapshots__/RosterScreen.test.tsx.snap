// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Roster Screen Tests ASSERT SOMETHING HERE 1`] = `
<View>
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "rgba(90 130 156 / 1.0)",
        "flexDirection": "row",
        "height": 50,
        "justifyContent": "flex-start",
        "paddingLeft": 15,
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "flex": 1,
          "flexGrow": 2,
        }
      }
    >
      <Text
        style={
          {
            "color": "white",
            "fontWeight": "bold",
          }
        }
      >
        Pos.
      </Text>
    </View>
    <View
      style={
        {
          "flex": 1,
          "flexGrow": 5,
        }
      }
    >
      <Text
        style={
          {
            "color": "white",
            "fontWeight": "bold",
          }
        }
      >
        Name
      </Text>
    </View>
    <View
      style={
        {
          "alignItems": "center",
          "flexGrow": 1,
        }
      }
    >
      <Text
        style={
          {
            "color": "white",
            "fontWeight": "bold",
          }
        }
      >
        Points
      </Text>
    </View>
  </View>
  <Modal
    animationType="slide"
    hardwareAccelerated={false}
    onRequestClose={[Function]}
    transparent={true}
    visible={false}
  />
  <RCTScrollView>
    <View>
      <View
        style={
          {
            "backgroundColor": "#fff",
            "flex": 1,
            "flexDirection": "row",
            "height": 50,
            "paddingTop": 30,
          }
        }
      >
        <View
          style={
            {
              "alignItems": "center",
              "flex": 1,
              "flexGrow": 2,
            }
          }
        >
          <Text
            style={{}}
          >
            G
          </Text>
        </View>
        <View
          style={
            {
              "flex": 1,
              "flexGrow": 5,
            }
          }
        >
          <Text
            style={{}}
          >
            Gilmore
          </Text>
        </View>
        <View
          style={
            {
              "flexGrow": 1,
            }
          }
        >
          <Text
            style={{}}
          >
            23
          </Text>
        </View>
      </View>
      <View
        style={
          {
            "backgroundColor": "#fff",
            "flex": 1,
            "flexDirection": "row",
            "height": 50,
            "paddingTop": 30,
          }
        }
      >
        <View
          style={
            {
              "alignItems": "center",
              "flex": 1,
              "flexGrow": 2,
            }
          }
        >
          <Text
            style={{}}
          >
            D
          </Text>
        </View>
        <View
          style={
            {
              "flex": 1,
              "flexGrow": 5,
            }
          }
        >
          <Text
            style={{}}
          >
            One
          </Text>
        </View>
        <View
          style={
            {
              "flexGrow": 1,
            }
          }
        >
          <Text
            style={{}}
          >
            23
          </Text>
        </View>
      </View>
      <View
        style={
          {
            "backgroundColor": "#fff",
            "flex": 1,
            "flexDirection": "row",
            "height": 50,
            "paddingTop": 30,
          }
        }
      >
        <View
          style={
            {
              "alignItems": "center",
              "flex": 1,
              "flexGrow": 2,
            }
          }
        >
          <Text
            style={{}}
          >
            D
          </Text>
        </View>
        <View
          style={
            {
              "flex": 1,
              "flexGrow": 5,
            }
          }
        >
          <Text
            style={{}}
          >
            Two
          </Text>
        </View>
        <View
          style={
            {
              "flexGrow": 1,
            }
          }
        >
          <Text
            style={{}}
          >
            23
          </Text>
        </View>
      </View>
      <View
        style={
          {
            "backgroundColor": "#fff",
            "flex": 1,
            "flexDirection": "row",
            "height": 50,
            "paddingTop": 30,
          }
        }
      >
        <View
          style={
            {
              "alignItems": "center",
              "flex": 1,
              "flexGrow": 2,
            }
          }
        >
          <Text
            style={{}}
          >
            LSM
          </Text>
        </View>
        <View
          style={
            {
              "flex": 1,
              "flexGrow": 5,
            }
          }
        >
          <Text
            style={{}}
          >
            Pole
          </Text>
        </View>
        <View
          style={
            {
              "flexGrow": 1,
            }
          }
        >
          <Text
            style={{}}
          >
            23
          </Text>
        </View>
      </View>
      <View
        style={
          {
            "backgroundColor": "#fff",
            "flex": 1,
            "flexDirection": "row",
            "height": 50,
            "paddingTop": 30,
          }
        }
      >
        <View
          style={
            {
              "alignItems": "center",
              "flex": 1,
              "flexGrow": 2,
            }
          }
        >
          <Text
            style={{}}
          >
            SSDM
          </Text>
        </View>
        <View
          style={
            {
              "flex": 1,
              "flexGrow": 5,
            }
          }
        >
          <Text
            style={{}}
          >
            McStickens
          </Text>
        </View>
        <View
          style={
            {
              "flexGrow": 1,
            }
          }
        >
          <Text
            style={{}}
          >
            23
          </Text>
        </View>
      </View>
      <View
        style={
          {
            "backgroundColor": "#fff",
            "flex": 1,
            "flexDirection": "row",
            "height": 50,
            "paddingTop": 30,
          }
        }
      >
        <View
          style={
            {
              "alignItems": "center",
              "flex": 1,
              "flexGrow": 2,
            }
          }
        >
          <Text
            style={{}}
          >
            FO
          </Text>
        </View>
        <View
          style={
            {
              "flex": 1,
              "flexGrow": 5,
            }
          }
        >
          <Text
            style={{}}
          >
            GetOff
          </Text>
        </View>
        <View
          style={
            {
              "flexGrow": 1,
            }
          }
        >
          <Text
            style={{}}
          >
            23
          </Text>
        </View>
      </View>
      <View
        style={
          {
            "backgroundColor": "#fff",
            "flex": 1,
            "flexDirection": "row",
            "height": 50,
            "paddingTop": 30,
          }
        }
      >
        <View
          style={
            {
              "alignItems": "center",
              "flex": 1,
              "flexGrow": 2,
            }
          }
        >
          <Text
            style={{}}
          >
            M
          </Text>
        </View>
        <View
          style={
            {
              "flex": 1,
              "flexGrow": 5,
            }
          }
        >
          <Text
            style={{}}
          >
            One
          </Text>
        </View>
        <View
          style={
            {
              "flexGrow": 1,
            }
          }
        >
          <Text
            style={{}}
          >
            23
          </Text>
        </View>
      </View>
      <View
        style={
          {
            "backgroundColor": "#fff",
            "flex": 1,
            "flexDirection": "row",
            "height": 50,
            "paddingTop": 30,
          }
        }
      >
        <View
          style={
            {
              "alignItems": "center",
              "flex": 1,
              "flexGrow": 2,
            }
          }
        >
          <Text
            style={{}}
          >
            M
          </Text>
        </View>
        <View
          style={
            {
              "flex": 1,
              "flexGrow": 5,
            }
          }
        >
          <Text
            style={{}}
          >
            Two
          </Text>
        </View>
        <View
          style={
            {
              "flexGrow": 1,
            }
          }
        >
          <Text
            style={{}}
          >
            23
          </Text>
        </View>
      </View>
      <View
        style={
          {
            "backgroundColor": "#fff",
            "flex": 1,
            "flexDirection": "row",
            "height": 50,
            "paddingTop": 30,
          }
        }
      >
        <View
          style={
            {
              "alignItems": "center",
              "flex": 1,
              "flexGrow": 2,
            }
          }
        >
          <Text
            style={{}}
          >
            A
          </Text>
        </View>
        <View
          style={
            {
              "flex": 1,
              "flexGrow": 5,
            }
          }
        >
          <Text
            style={{}}
          >
            McGavin
          </Text>
        </View>
        <View
          style={
            {
              "flexGrow": 1,
            }
          }
        >
          <Text
            style={{}}
          >
            23
          </Text>
        </View>
      </View>
      <View
        style={
          {
            "backgroundColor": "#fff",
            "flex": 1,
            "flexDirection": "row",
            "height": 50,
            "paddingTop": 30,
          }
        }
      >
        <View
          style={
            {
              "alignItems": "center",
              "flex": 1,
              "flexGrow": 2,
            }
          }
        >
          <Text
            style={{}}
          >
            A
          </Text>
        </View>
        <View
          style={
            {
              "flex": 1,
              "flexGrow": 5,
            }
          }
        >
          <Text
            style={{}}
          >
            Two
          </Text>
        </View>
        <View
          style={
            {
              "flexGrow": 1,
            }
          }
        >
          <Text
            style={{}}
          >
            23
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
